bot/cogs/information.py

```diff
@@ -114,17 +144,27 @@ async def server_info(self, ctx: Context) -> None:
         channel_counts = "\n".join(channel_type_list).strip()
 
         # How many channels are for staff only?
-        everyone_role = ctx.guild.roles[0]
-        hidden_channels = 0
-
-        for channel in ctx.guild.channels:
-            overwrites = channel.overwrites_for(everyone_role)
-            if overwrites.is_empty():
-                continue
-
-            for perm, value in overwrites:
-                if perm == 'read_messages' and value is False:
-                    hidden_channels += 1
+        # We need to know two things about a channel:
+        # - Does the @everyone role have explicit read deny permissions?
+        # - Do staff roles have explicit read allow permissions?
+        #
+        # If the answer to both of these questions is yes, it's a staff channel.
+        helpers = ctx.guild.get_role(constants.Roles.helpers)
+        moderators = ctx.guild.get_role(constants.Roles.moderators)
+        admins = ctx.guild.get_role(constants.Roles.admins)
+        everyone = ctx.guild.roles[0]
+
+        # Let's build some lists of channels.
+        everyone_denied = self._get_channels_where_role_cannot_read(ctx, everyone)
+        staff_allowed = more_itertools.flatten([
+            self._get_channels_where_role_can_read(ctx, admins),       # Admins has explicit read message allow
+            self._get_channels_where_role_can_read(ctx, moderators),   # Moderators has explicit read message allow
+            self._get_channels_where_role_can_read(ctx, helpers),      # Helpers has explicit read message allow
+        ])
```

Comment:
It is probably insignificant, but it strikes me that calling the function 4 times is inefficient. This could all be determined within one pair of iterations, though perhaps at the cost of readability or cleanliness.

Reply:
It definitely is, I did it for readability reasons. There may be an approach that finds a better balance between readability and optimization